### ISSUE TITLE
[$1000] No new marker if tab opened via link to the specific chat

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#12598]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Issue was found when executing #12169 
## Action Performed:
1. Open two sessions one as User A and one as User B
2. Make sure User B is navigate to the chat with User A and save the URL
3. Close User B tab (or kill app)
4. Create message as User A on chat with User B and wait 10 seconds
5. Open tab (or app) as User B using the URL saved from the second step (on native use a deep link)
6. See that new marker is set correctly
7. Verify that the chat in the LHN is marked as read

## Expected Result:
new marker is set correctly

## Actual Result:
there is no new marker

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App - could not test as it requires copying URL

**Version Number:** 1.2.25-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/200882354-870b6de6-3361-4598-b3c5-009f12034a78.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** Applause internal team
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a1559dc1509cad8c</li>
        <li>Upwork Job ID: 1622812603121872896</li>
        <li>Last Price Increase: 2023-02-07</li>
    </ul>
</details>

**Comment by dylanexpensify** [comment]
cc @marcaaron curious for your eyes on this given [this PR](https://github.com/Expensify/App/pull/12169)

**Comment by dylanexpensify** [comment]
reviewing today!

**Comment by dylanexpensify** [comment]
reviewing today mel!

**Comment by dylanexpensify** [comment]
Upwork: https://www.upwork.com/jobs/~01d9c17d63e3cc48ee

**Comment by MonilBhavsar** [comment]
I'm able to reproduce the issue. Looking into it

**Comment by MonilBhavsar** [comment]
Upon investigation, looks like I'm not receiving Pusher event at all when report is not visible and there are some Pusher related errors in the log. I'll sort it out and update here

**Comment by dylanexpensify** [comment]
Thanks @MonilBhavsar!

**Comment by puneetlath** [comment]
@MonilBhavsar  this is one of the oldest issues in the /App repo. To help us clear out the large backlog of bugs, can you:  
- Decide whether any proposals currently meet our guidelines and can be approved as-is 
- For any that can't, please take this issue internal and treat it as one of your highest priorities 
- If you have any questions, don't hesitate to start a discussion in #bug-zero

**Comment by dylanexpensify** [comment]
reviewing today!

**Comment by dylanexpensify** [comment]
@MonilBhavsar thoughts on what Puneet posted? If we're good with external, I'll move it to $1k price today

**Comment by MonilBhavsar** [comment]
I've been working on this issue only. Will post my findings shortly

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/pages/home/ReportScreen.js b/src/pages/home/ReportScreen.js
index 972dc0210..d3ba9e03d 100644
--- a/src/pages/home/ReportScreen.js
+++ b/src/pages/home/ReportScreen.js
@@ -216,6 +216,7 @@ class ReportScreen extends React.Component {
 
         // There are no reportActions at all to display and we are still in the process of loading the next set of actions.
         const isLoadingInitialReportActions = _.isEmpty(this.props.reportActions) && this.props.report.isLoadingReportActions;
+        const isLoadingReportData = this.props.isLoadingReportData;
         return (
             <Freeze
                 freeze={this.props.isSmallScreenWidth && this.props.isDrawerOpen}
@@ -279,7 +280,7 @@ class ReportScreen extends React.Component {
                                 this.setState({skeletonViewContainerHeight});
                             }}
                         >
-                            {(this.isReportReadyForDisplay() && !isLoadingInitialReportActions) && (
+                            {(this.isReportReadyForDisplay() && !isLoadingInitialReportActions && !isLoadingReportData) && (
                                 <>
                                     <ReportActionsView
                                         reportActions={this.props.reportActions}
@@ -303,7 +304,7 @@ class ReportScreen extends React.Component {
 
                             {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent rendering the report while they are loading then
                             we'll unnecessarily unmount the ReportActionsView which will clear the new marker lines initial state. */}
-                            {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions) && (
+                            {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions || isLoadingReportData) && (
                                 <ReportActionsSkeletonView
                                     containerHeight={this.state.skeletonViewContainerHeight}
                                 />
@@ -353,5 +354,8 @@ export default compose(
         personalDetails: {
             key: ONYXKEYS.PERSONAL_DETAILS,
         },
+        isLoadingReportData: {
+            key: ONYXKEYS.IS_LOADING_REPORT_DATA,
+        },
     }),
 )(ReportScreen);
```

### Details
`<ReportActionsView/>` calls `ReportUtils.isUnread()` on the old data (locally stored) and does not wait for `OpenApp`. And the old data have `maxSequenceNumber === lastReadSequenceNumber` Thus there is no new mark indicator.
In my proposal the fix is to not call the `<ReportActionsView/>` while the app is still loading.

**Comment by sobitneupane** [comment]
> I've been working on this issue only. Will post my findings shortly

@MonilBhavsar It looks like you are handling it internally. Should I review the proposal?

**Comment by MonilBhavsar** [comment]
Not necessarily Internal, but we don't want to have this issue stay still. So I'm also looking into it to fix this asap.

**Comment by MonilBhavsar** [comment]
@marcaaron since this from https://github.com/Expensify/App/pull/12169. A quick question - Why the QA steps mention to wait for 10 seconds and then reopen the tab for User B.
I think this is opposed to what is expected, no?
The callback you added won't  triggered if we're past 10 seconds
https://github.com/Expensify/App/blob/39a68d1b56161d378eabf66578aefbd28b1fb8ac/src/libs/actions/Report.js#L1332

**Comment by sobitneupane** [comment]
@s77rt Thanks for your proposal.

Your proposal solves the issue but will create other issue. When user reloads page being Offline, user won't see previously loaded messages.

**Comment by s77rt** [comment]
### Proposal (Update)
```diff
diff --git a/src/pages/home/ReportScreen.js b/src/pages/home/ReportScreen.js
index 972dc0210..84fbb38f3 100644
--- a/src/pages/home/ReportScreen.js
+++ b/src/pages/home/ReportScreen.js
@@ -216,6 +216,7 @@ class ReportScreen extends React.Component {
 
         // There are no reportActions at all to display and we are still in the process of loading the next set of actions.
         const isLoadingInitialReportActions = _.isEmpty(this.props.reportActions) && this.props.report.isLoadingReportActions;
+        const isLoadingReportData = this.props.isLoadingReportData && !this.props.network.isOffline;
         return (
             <Freeze
                 freeze={this.props.isSmallScreenWidth && this.props.isDrawerOpen}
@@ -279,7 +280,7 @@ class ReportScreen extends React.Component {
                                 this.setState({skeletonViewContainerHeight});
                             }}
                         >
-                            {(this.isReportReadyForDisplay() && !isLoadingInitialReportActions) && (
+                            {(this.isReportReadyForDisplay() && !isLoadingInitialReportActions && !isLoadingReportData) && (
                                 <>
                                     <ReportActionsView
                                         reportActions={this.props.reportActions}
@@ -303,7 +304,7 @@ class ReportScreen extends React.Component {
 
                             {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent rendering the report while they are loading then
                             we'll unnecessarily unmount the ReportActionsView which will clear the new marker lines initial state. */}
-                            {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions) && (
+                            {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions || isLoadingReportData) && (
                                 <ReportActionsSkeletonView
                                     containerHeight={this.state.skeletonViewContainerHeight}
                                 />
@@ -353,5 +354,8 @@ export default compose(
         personalDetails: {
             key: ONYXKEYS.PERSONAL_DETAILS,
         },
+        isLoadingReportData: {
+            key: ONYXKEYS.IS_LOADING_REPORT_DATA,
+        },
     }),
 )(ReportScreen);
```

### Update
@sobitneupane I'm not able to reproduce the issue you just described, on reload the page does not load at all.
However I updated my proposal to what I suppose will fix the scenario you just described

**Comment by s77rt** [comment]
> @marcaaron since this from #12169. A quick question - Why the QA steps mention to wait for 10 seconds and then reopen the tab for User B. I think this is opposed to what is expected, no? The callback you added won't triggered if we're past 10 seconds
> 
> https://github.com/Expensify/App/blob/39a68d1b56161d378eabf66578aefbd28b1fb8ac/src/libs/actions/Report.js#L1332

@MonilBhavsar I was able to reproduce the bug without waiting that long. I think what the reporter meant is to wait for the message to get actually sent, because the backend server is a little buggy and sometimes it takes up several seconds to get my message sent. Having specified 10 seconds may be just a coincidence.

**Comment by marcaaron** [comment]
>the backend server is a little buggy and sometimes it takes up several seconds to get my message sent

@s77rt No, that's not correct.

@MonilBhavsar It's because the logic for this needs improvement as discussed [here](https://github.com/Expensify/Expensify/issues/228194). We wait 10 seconds to make sure that we are not handling a comment read into storage as if it were "new" twice. This is very uncommon when using the app normally, but much more likely when testing.

**Comment by marcaaron** [comment]
Gonna drop the `Help Wanted` from this issue I think we should bring this issue to Slack to discuss further (same goes for any other open issues related to new marker or unread indicator inconsistencies). These are tough problems to figure out and may require access to our backend code.

**Comment by dylanexpensify** [comment]
Catching up now that I'm back from OOO

**Comment by dylanexpensify** [comment]
@marcaaron can we link where the convo is happening you alluded to [here](https://github.com/Expensify/App/issues/12598#issuecomment-1325673998)? Should we put HOLD on this? Thanks!

**Comment by MonilBhavsar** [comment]
Getting to this today

**Comment by dylanexpensify** [comment]
thanks @MonilBhavsar !

**Comment by MonilBhavsar** [comment]
I see that we're going to update New report action pusher event in the Reliable Notification doc. So I think we should hold this issue for the doc. cc @Julesssss @arosiclair

**Comment by marcaaron** [comment]
@MonilBhavsar What part of this needs to be on hold? I'd suggest bringing this one to Slack with a summary so we can discuss further and get it unblocked.

**Comment by dylanexpensify** [comment]
Friendly bump @MonilBhavsar 😄

**Comment by dylanexpensify** [comment]
agreed - @MonilBhavsar thoughts?

**Comment by dylanexpensify** [comment]
[posted](https://expensify.slack.com/archives/C01GTK53T8Q/p1670922665510229) cc @trjExpensify !

**Comment by MonilBhavsar** [comment]
Sorry it got slipped. Commented on the thread on working on the fix

**Comment by marcaaron** [comment]
@arosiclair I agree that this doesn't seem to have anything to do with Pusher.

**Comment by trjExpensify** [comment]
👋 What's the latest here?

**Comment by dylanexpensify** [comment]
@MonilBhavsar any updates?

**Comment by MonilBhavsar** [comment]
I'm willing to put this on hold since we're refactoring the code around this logic. I have dropped message in slack. I'll update it accordingly here

**Comment by dylanexpensify** [comment]
@MonilBhavsar when you have a moment, can you summarize the slack convo here and the decision we made?

**Comment by MonilBhavsar** [comment]
Yes, on it

**Comment by MonilBhavsar** [comment]
Discussing it here https://expensify.slack.com/archives/C01GTK53T8Q/p1671704053216749

**Comment by MonilBhavsar** [comment]
Flagged over weekend. Still discussing...

**Comment by MonilBhavsar** [comment]
Looks like everyone agrees with P/S statement. Will put up a PR to fix this issue

**Comment by JmillsExpensify** [comment]
Moving this into the sequence numbers project given that this is where we're undergoing a large unreads refactor at the moment. Also putting this issue on hold as a result.

**Comment by MonilBhavsar** [comment]
On hold

**Comment by JmillsExpensify** [comment]
Still on hold, Melvin.

**Comment by JmillsExpensify** [comment]
Just realized that I forgot to update the label given that this is on hold.

**Comment by JmillsExpensify** [comment]
Removed the hold now that sequence numbers are now deprecated. I'll retest as well.

**Comment by JmillsExpensify** [comment]
Hmm, looks like this unread bug is still with us. cc @chiragsalian should we take these trailing unread bugs internal as part of clean-up for this project?

**Comment by MonilBhavsar** [comment]
Agree, still reproducible

**Comment by amyevans** [comment]
If we want to keep this internal, another engineer should feel free to grab it, I'm not sure I'll have time to look into it until Monday.

**Comment by MelvinBot** [comment]
@JmillsExpensify, @sobitneupane, @MonilBhavsar Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@JmillsExpensify, @sobitneupane, @MonilBhavsar Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@JmillsExpensify, @sobitneupane, @MonilBhavsar Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01a1559dc1509cad8c

**Comment by MelvinBot** [comment]
Current assignee @JmillsExpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @sobitneupane is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @MonilBhavsar is eligible for the External assigner, not assigning anyone new.

**Comment by JmillsExpensify** [comment]
Marked external again so that we can get some new ideas/proposals on how to solve it.

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $1000

**Comment by harshjanipa** [comment]
@JmillsExpensify @MelvinBot Question What the new marker set should look like ? answer would helps to understand the issue.

**Comment by JmillsExpensify** [comment]
Oh the "new" marker already exists. Just mark any message unread in the app and you'll see a "new" line like below. 

![Screenshot 2023-02-07 at 10 51 32](https://user-images.githubusercontent.com/1127863/217310241-7083ae0e-0591-4130-9c36-9039e23c94c9.png)

**Comment by sobitneupane** [comment]
@Ashwanikhurana Thanks for your proposal.

Can you please repost the proposal using the [proposal template](https://github.com/Expensify/App/blob/main/contributingGuides/PROPOSAL_TEMPLATE.md)? 

You can follow [our guideline](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job) to know how you are supposed to propose a solution.

**Comment by harshjanipa** [comment]
@JmillsExpensify @sobitneupane 

## Proposal

### Please re-state the problem that we are trying to solve in this issue.

No New Marker when User A send message to User B if User B is not active(not logged in or open app using URL).

### What is the root cause of that problem?

I have forked and downloaded the APP local and traced the code and found 

`const shouldDisplayNewMarker = reportAction.reportActionID === this.props.newMarkerReportActionID;`

Above logic only compares ID for newMaker to appear. there is not check is the message is new or not.

### What changes do you think we should make in order to solve the problem?

add property  isNewMessage  : true with each message user sends.
 depends upon User experience mark isNewMessage  : false .
e.g(onFocus) onBlur or more in 

componentDidMount or useEffect() 
reason if component is already mounted means userB is active(window is open) and when new message receives mark message 
isNewMessage : false, when receiving message from another user, where messageArray is increased.

How to know if message array from another user have increased ? 
Answer : using this.setState((**prevState**) => {})

**Comment by sobitneupane** [comment]
@harshjanipa Thanks for your proposal. 
 But I am not sure if I understood the root cause of the problem you stated. Can you please explain a bit about it? 
 
 It works as expected when a user is active and receives a message. But not if the user navigates via link. What's the reason for this behavior?

**Comment by MelvinBot** [comment]
@JmillsExpensify, @sobitneupane, @MonilBhavsar Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@JmillsExpensify, @sobitneupane, @MonilBhavsar Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by sobitneupane** [comment]
Waiting for proposals.

**Comment by s77rt** [comment]
@sobitneupane Would you please check this [proposal](https://github.com/Expensify/App/issues/12598#issuecomment-1324237410) again.
I believe this issue will be solved by the above proposal in combination with this proposal [here](https://github.com/Expensify/App/issues/15019#issuecomment-1426067462).

___

Sorry for not providing much info in this comment, I think not much has changed since the initial proposal.

**Comment by abdulrahuman5196** [comment]
@sobitneupane This proposal could solve this issue as well - https://github.com/Expensify/App/issues/15019#issuecomment-1425434440

**Comment by s77rt** [comment]
@sobitneupane partly the same. After that issue get fixed we will have to apply [this](https://github.com/Expensify/App/issues/12598#issuecomment-1324641102) to actually fix this issue.

**Comment by sobitneupane** [comment]
@MonilBhavsar I think we should either hold this issue or combine this issue with https://github.com/Expensify/App/issues/15019 one.

**Comment by JmillsExpensify** [comment]
I commented in Slack, but I favor combining this issue in the other issue and making sure that we solve both in the same go. This was also reported internally, so no issues there.

**Comment by JmillsExpensify** [comment]
Thoughts on that last comment @sobitneupane? I think we should close this issue out.

**Comment by sobitneupane** [comment]
Yes, we can close this issue. Let's make sure we solve this issue as well while handling https://github.com/Expensify/App/issues/15019 issue.

cc: @thesahindia

**Comment by JmillsExpensify** [comment]
Done! Added the steps to the linked issue. Thanks everyone!

**Comment by MonilBhavsar** [comment]
I agree with this. but I would say, let's add the workflow of this issue to the other issue and consider it while fixing it.
Or, create a new tracking issue describing all bugs with new marker line and close duplicate issues

**Comment by JmillsExpensify** [comment]
![Screenshot 2023-02-16 at 10 44 28](https://user-images.githubusercontent.com/1127863/219431440-4728407a-e0ff-44de-a18b-4c7b492bcc9f.png)

@MonilBhavsar the case for this issue was already added to the linked issue. Screenshot from that OP for clarity.

**Comment by JmillsExpensify** [comment]
Separately from that, I love the idea of creating a tracking issue for the new marker line. We're experiencing multiple issues at this point and we're not going to holistically resolve until we look at everything holistically.
